### ISSUE TITLE
add Docuseal (requires text/plain support or hack)

### ISSUE THREAD

**Issue by meonkeys** [bastienwirtz/homer#847]
Docuseal [now exposes its version at `/version`](https://github.com/docusealco/docuseal/pull/400#issuecomment-2555514960), but the response is a `text/plain` string like `1.8.3a`, not JSON. Can Homer somehow use this as-is?

I tried creating a custom service / smart card like the others I've made, but I get `SyntaxError: JSON.parse: unexpected non-whitespace character after JSON data at line 1 column 4 of the JSON data` in the console.

I tried manually adding double quotes around the response, making it `"1.8.3a"`...

```javascript
this.versionstring = '"' + response + '"';
```

then Homer accepts the string, but it prints `Version "1.8.3a"` on the smart card (with double quotes).

**Comment by bastienwirtz** [comment]
Hello @meonkeys !

I assume you got the parse error using the `this.fetch` method ? did you try using it with the third parameter to false?
```js
// fetch: function (path, init, json = true) -> the json parameter will activate or not the json parsing
this.fetch('http://your-url', null, false);
```

**Comment by meonkeys** [comment]
Oh nice, that helps, thank you.

Looks like `fetch()` in the "service" mixin needs a little patch, though:

```diff
diff --git a/src/mixins/service.js b/src/mixins/service.js
index b4fd2ad..7051a8f 100644
--- a/src/mixins/service.js
+++ b/src/mixins/service.js
@@ -56,7 +56,7 @@ export default {
           throw new Error(`Ping: target not available (${response.status} error)`);
         }
 
-        return json ? response.json() : response;
+        return json ? response.json() : response.text();
       });
     },
   },
```

Without that I'm getting `Version "[object Response]"` on the smart card.

Want a PR for that? Separate PR, I assume?

**Comment by meonkeys** [comment]
see #850

**Comment by bastienwirtz** [comment]
Good catch @meonkeys ! The response body wihtout json parsing has never been used (the ping component use the third parameter to false, but it does not use the response).
And no worries for the the PR, in the same one is fine.
